pkg/hierarchy/manager.go

```diff
@@ -52,26 +72,38 @@ func (m *Manager[CQ, C]) UpdateClusterQueueEdge(name, parentName string) {
 }
 
 func (m *Manager[CQ, C]) DeleteClusterQueue(name string) {
-	if cq, ok := m.ClusterQueues[name]; ok {
+	if cq, ok := m.clusterQueues[name]; ok {
 		m.detachClusterQueueFromParent(cq)
-		delete(m.ClusterQueues, name)
+		delete(m.clusterQueues, name)
 	}
 }
 
 func (m *Manager[CQ, C]) AddCohort(cohortName string) {
-	oldCohort, ok := m.Cohorts[cohortName]
+	oldCohort, ok := m.cohorts[cohortName]
 	if ok && oldCohort.isExplicit() {
 		return
 	}
 	if !ok {
-		m.Cohorts[cohortName] = m.cohortFactory(cohortName)
+		m.cohorts[cohortName] = m.cohortFactory(cohortName)
 	}
-	m.Cohorts[cohortName].markExplicit()
+	m.cohorts[cohortName].markExplicit()
+}
+
+func (m *Manager[CQ, C]) GetCohort(name string) C {
+	return m.cohorts[name]
+}
+
+func (m *Manager[CQ, C]) GetCohortsCopy() map[string]C {
```

Comment:
```suggestion
func (m *Manager[CQ, C]) Cohorts() map[string]C {
```
nit, no need for Get, and we can probably implement with maps.Clone

Reply:
Updated all getter names and use maps.Clone for cloning